deeppavlov/models/nemo/tts.py

```diff
@@ -0,0 +1,210 @@
+            bos_id: Label position of beginning of string symbol.
+            eos_id: Label position of end of string symbol.
+            lowercase: Whether to convert all uppercase characters in a text batch into lowercase characters.
+
+        """
+        parser = parsers.make_parser(labels, do_lowercase=lowercase)
+        self.texts = collections.Text(text_batch, parser)
+        self.bos_id = bos_id
+        self.eos_id = eos_id
+
+
+class TextDataLayer(CustomDataLayerBase):
+    @property
+    @add_port_docs()
+    def output_ports(self) -> Dict[str, NeuralType]:
+        return {
+            'texts': NeuralType(('B', 'T'), LabelsType()),
+            "texts_length": NeuralType(tuple('B'), LengthsType())
+        }
+
+    def __init__(self, *,
+                 text_batch: List[str],
+                 labels: List[str],
+                 batch_size: int = 32,
+                 bos_id: Optional[int] = None,
+                 eos_id: Optional[int] = None,
+                 pad_id: Optional[int] = None,
+                 **kwargs) -> None:
+        """A simple Neural Module for loading text data.
+
+        Args:
+            text_batch: Texts to be used for speech synthesis.
+            labels: List of string labels to use when to str2int translation.
+            batch_size: How many strings per batch to load.
+            bos_id: Label position of beginning of string symbol. If None is initialized as `len(labels)`.
+            eos_id: Label position of end of string symbol. If None is initialized as `len(labels) + 1`.
+            pad_id: Label position of pad symbol. If None is initialized as `len(labels) + 2`.
+
+        """
+        len_labels = len(labels)
+        if bos_id is None:
+            bos_id = len_labels
+        if eos_id is None:
+            eos_id = len_labels + 1
+        if pad_id is None:
+            pad_id = len_labels + 2
+
+        dataset = TextDataset(text_batch=text_batch, labels=labels, bos_id=bos_id, eos_id=eos_id)
+
+        dataloader = torch.utils.data.DataLoader(dataset=dataset, batch_size=batch_size,
+                                                 collate_fn=partial(self._collate_fn, pad_id=pad_id))
+        super(TextDataLayer, self).__init__(dataset, dataloader, **kwargs)
+
+    @staticmethod
+    def _collate_fn(batch: Tuple[Tuple[Tensor], Tuple[Tensor]], pad_id: int) -> Tuple[Tensor, Tensor]:
+        """Collates batch of texts.
+
+        Args:
+            batch: A tuple of tuples of audio signals and signal lengths.
+            pad_id: Label position of pad symbol.
+
+        Returns:
+            texts: Padded texts tensor.
+            texts_len: Text lengths tensor.
+
+        """
+        texts_list, texts_len = zip(*batch)
+        max_len = max(texts_len)
+        max_len = pad_to(max_len, 8)
+
+        texts = torch.empty(len(texts_list), max_len, dtype=torch.long)
+        texts.fill_(pad_id)
+
+        for i, s in enumerate(texts_list):
```

Comment:
s is probably not unambigous here

Reply:
"not unambigous" == "ambigous"?) Got it `s` -> `text`